### PR TITLE
User can delete post

### PR DESCRIPTION
**Pull request by erikwesterberg**
PR for delete post feature
```
As a user,
In order to get rid of a wrong post,
I would like to able to delete it.
```
PT: https://www.pivotaltracker.com/story/show/167366428
FEATURE GIF: https://recordit.co/y6lEsEw2rt

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -11,6 +11,9 @@
           <td><%= post.title %></td>
           <td><%= post.text %></td>
           <td><%= post.user.name %></td>
+          <td><%= link_to 'Delete', post_path(post),
```
Comment (oliverochman) [inline app/views/posts/index.html.erb]: Should anyone be able to delete a post?

File: features/user_can_delete_post.feature [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  Background:
+    Given the following post exists
+    | title        | text          | 
+    | Codeacademy  | Smooth-ride   | 
+    And the following user exists
+    | name    | email            | password     |
+    | Fat Bob | fatbob@gmail.com | ilovecatfood |
+    And I am logged in as "Fat Bob"  
+
+  Scenario: Reader can open up posts
```
Comment (oliverochman) [inline features/user_can_delete_post.feature]: You should have a scenario where another user is logged in and cant see the delete button

File: app/views/posts/index.html.erb (hunk 1 above)
Comment (pedbr) [reply app/views/posts/index.html.erb]: Now user can only delete own posts

File: features/user_can_delete_post.feature (hunk 2 above)
Comment (pedbr) [reply features/user_can_delete_post.feature]: Test created and passing

File: features/user_can_delete_post.feature [hunk 3]
```diff
@@ -0,0 +1,28 @@
+Feature:
+"As a user,
+In order to get rid of a wrong post,
+I would like to able to delete it."
+
+  Background:
+    Given the following user exists
+    | name     | email             | password     |
+    | Fat Bob  | fatbob@gmail.com  | ilovecatfood |
+    | Slim Jim | slimjum@gmail.com | password123  |
```
Comment (tochman) [inline features/user_can_delete_post.feature]: I think you can simplify this even more. You don't need the email and password in the test. You can allow FactoryBot to set that.

Comment (tochman) [reply features/user_can_delete_post.feature]: Just keep the name

Comment (pedbr) [reply features/user_can_delete_post.feature]: Done!

File: app/views/posts/index.html.erb (hunk 1 above)
Comment (oliverochman) [reply app/views/posts/index.html.erb]: It's good that you added that `if` statement to the controller action. But personally, I feel that it is not good UX to show the user a button for deleting a post when they cant delete it.

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: app/views/posts/index.html.erb (hunk 1 above)
Comment (pedbr) [reply app/views/posts/index.html.erb]: After the work we had building that if statement!! It hurts but you're totally right! It's fixed now! Only the post owner can see the Delete button!

**Review by oliverochman — APPROVED** [review]

**Pull request merged**